cmd/queue/main.go

```diff
@@ -106,22 +107,34 @@ func connectStatSink() {
 	autoscalerEndpoint := fmt.Sprintf("ws://%s.%s.svc.cluster.local:%s",
 		servingAutoscaler, system.Namespace, servingAutoscalerPort)
 	logger.Infof("Connecting to autoscaler at %s.", autoscalerEndpoint)
-	for {
-		// TODO: use exponential backoff here
-		time.Sleep(time.Second)
 
+	// Will retry connecting to the autoscaler for roughly 1.5 minutes.
+	err := wait.ExponentialBackoff(wait.Backoff{
+		Duration: 100 * time.Millisecond,
+		Factor:   1.3,
+		Steps:    20,
+		Jitter:   0.5,
+	}, func() (bool, error) {
 		dialer := &websocket.Dialer{
 			HandshakeTimeout: 3 * time.Second,
 		}
 		conn, _, err := dialer.Dial(autoscalerEndpoint, nil)
 		if err != nil {
 			logger.Error("Retrying connection to autoscaler.", zap.Error(err))
-		} else {
-			logger.Info("Connected to stat sink.")
-			statSink = conn
-			waitForClose(conn)
+			return false, nil
 		}
+
+		logger.Info("Connected to autoscaler.")
+		statSink = conn
+		return true, nil
+	})
+
+	// Exit iff a connection could not be established in the timeout boundaries.
+	if err == wait.ErrWaitTimeout {
+		logger.Fatal("Was not able to connect to autoscaler. Aborting.")
```

Comment:
/cc @josephburnett 

Ok to exit queue-proxy if it can't connect to the autoscaler?